### PR TITLE
implemented basic Elm template

### LINKED ISSUE #1: Elm Template for Embark
### Outline

The goal of this task is to create a ready to use Elm template for [Embark](https://embark.status.im/docs/). The template should have:

* The necessary packages in package.json for Elm
* A *simple* sample code using Elm
* Additions/Modifications in the (embark's) webpack to support Elm

See an example of a existing template with a custom webpack config such as the typescript template available [here](https://github.com/embark-framework/embark-typescript-template). 

This task is meant for the embark version currently in the develop branch, to install:

```
git clone git@github.com:embark-framework/embark.git
cd embark
npm install
npm link
```

starting point:
```
embark new MyApp
cd MyApp
embark eject-webpack
```

### Acceptance Criteria
* Running this template with `embark run` should result in a usable dapp using Elm as minimal starting point to start using a dapp using Elm and embark.
* It has to use embark and its pipeline (configured in `embark.json` (not elm cmd line tools) and using the embark's webpack config file (e.g found [here](https://github.com/embark-framework/embark-ember-template/blob/master/webpack.config.js)).

### PR DESCRIPTION
**Pull request by evgeniuz**
Fixes #1.

### CODE REVIEW

**Comment by evgeniuz** [comment]
BTW, first commit (https://github.com/embark-framework/embark-elm-template/pull/2/commits/70605d95ee05564fc1e68376aedf3144f108edf1) in the pull request is the state right after `embark new` and `embark eject`, so you can see diff starting from that commit to see what's changed from JS template.

**Comment by evgeniuz** [comment]
Is there anything I need to improve here?

**Comment by StatusSceptre** [comment]
@jrainville @iurimatias wdyt? Have you looked at this and, if so, can we merge and payout the bounty? This has been lingering for some time now.

File: config/contracts.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+module.exports = {
+  // default applies to all environments
+  default: {
+    // Blockchain node to deploy the contracts
+    deployment: {
+      host: "localhost", // Host of the blockchain node
+      port: 8545, // Port of the blockchain node
+      type: "rpc" // Type of connection (ws or rpc),
```
Comment (jrainville) [inline config/contracts.js]: We should switch to port 8546 and type `ws` as `rpc` is deprecated.

**Review by jrainville — COMMENTED** [review]
Sorry, this completely passed under the radar with Cryptolife and Devcon. 
I'll test in a few. Also, just one small thing to change.

**Comment by jrainville** [comment]
Just tested with embark 3.2.3 and I have the following error:
```
./app/elm/Main.elm
Module build failed (from ./node_modules/elm-webpack-loader/index.js):
Error: Compiler process exited with error Compilation failed
-- BAD JSON ----------------------------------------------------------- elm.json

The "source-directories" in your elm.json lists the following directory:

    src

I cannot find that directory though! Is it missing? Is there a typo?


    at ChildProcess.<anonymous> (C:\dev\test\elm-test\node_modules\node-elm-compiler\index.js:149:27)
    at emitTwo (events.js:126:13)
    at ChildProcess.emit (events.js:214:7)
    at ChildProcess.cp.emit (C:\dev\test\elm-test\node_modules\cross-spawn\lib\enoent.js:40:29)
    at maybeClose (internal/child_process.js:925:16)
    at Socket.stream.socket.on (internal/child_process.js:346:11)
    at emitOne (events.js:116:13)
    at Socket.emit (events.js:211:7)
    at Pipe._handle.close [as _onclose] (net.js:557:12)
 @ ./app/js/index.js 6:0-34 7:10-13
 @ multi ./app/js/index.js
```

Steps:
1- `embark new --template https://github.com/evgeniuz/embark-elm-template elm-test`
2- `cd elm-test`
3- `embark run --nodashboard`
4- Fails while writing files.

**Comment by evgeniuz** [comment]
@jrainville Please, take a look at https://github.com/embark-framework/embark-elm-template/pull/3, based this on top of newer embark version.

**Comment by frankchen07** [comment]
hey @jrainville - Frank from Gitcoin here - helping @evgeniuz check in for this PR. Thanks!

**Pull request closed without merging**